src/stratigraphy/groundwater/groundwater_extraction.py

```diff
@@ -285,12 +482,24 @@ def extract_groundwater(self, terrain_elevation: Elevation | None) -> list[Groun
         """
         for page in self.doc:
             lines = extract_text_lines(page)
-            page_number = page.number + 1  # page.number is 0-based
-
-            found_groundwater = (
-                self.get_groundwater_near_key(lines, page_number)
-                # or XXXX # Add other techniques here
-            )
+            page_number = page.number + 1  # NOTE: page.number is 0-based
+
+            found_groundwater = self.get_groundwater_near_key(lines, page_number)
+            if not found_groundwater:
+                logger.info("No groundwater found near the key on page %s.", page_number)
+
+                ### Extract groundwater from illustration
+                if self.is_searching_groundwater_illustration:
+                    found_groundwater, confidence_list = self.get_groundwater_from_illustration(
+                        lines, page_number, terrain_elevation
+                    )
+                    if found_groundwater:
+                        logger.info("Confidence list: %s", confidence_list)
```

Comment:
Is this helpful logging according to you, @stijnvermeeren-swisstopo? In a previous iteration, we removed the logging in case no groundwater was found. We can also remove the logging when groundwater is found, as I mostly use it for debugging purposes.